edgelet/edgelet-http/src/util/incoming.rs

```diff
@@ -34,6 +51,60 @@ impl Stream for Incoming {
                     Some((StreamSelector::Tcp(stream), IncomingSocketAddr::Tcp(addr)))
                 })
             }
+            #[cfg(unix)]
+            Incoming::Tls(ref mut listener, ref mut acceptor, ref mut connections) => {
+                // check if we have a tcp connection and if we do then kick off TLS handshake
+                // and store the future representing that operation in "connections"
+                if let Err(err) = listener.poll_accept().map(|val| {
+                    if let Async::Ready((tcp_stream, addr)) = val {
+                        connections
+                            .lock()
+                            .expect("Unable to lock the connection mutex")
+                            .push((acceptor.accept(tcp_stream), IncomingSocketAddr::Tcp(addr)));
+                    }
+                }) {
+                    return Err(err);
+                }
+
+                // Take a read lock on our pending connection list returning a tuple representing an index
+                // + the state on the poll
+                let mut connections = connections
+                    .lock()
+                    .expect("Unable to lock the connection mutex");
+                let val = connections
+                    .iter_mut()
+                    .map(|(fut, _)| fut.poll())
+                    .enumerate()
+                    .find(|(_, result)| match result {
+                        Ok(v) => v.is_ready(),
+                        Err(_) => true,
+                    });
+
+                // Validate that the poll is ready, and remove that value. Then return the connection.
+                // If no connections are available in connection manager, return Async::NotReady
+                match val {
+                    Some((i, result)) => {
+                        let (_, addr) = connections.remove(i);
+                        match result {
+                            Ok(Async::Ready(tls_stream)) => {
+                                #[cfg(not(windows))]
+                                {
+                                    Async::Ready(Some((StreamSelector::Tls(tls_stream), addr)))
+                                }
+                                #[cfg(windows)]
```

Comment:
Since the enum variant `Incoming::Tls` is not available on Windows, I am wondering why this code compiles on Windows causing the "variant too large" clippy error. Clippy doesn't care for `cfg` annotations?

Reply:
Well the StreamSelector was not using cfg(Unix) for the TLS stream since none of that code relied on OpenSSL. I suppose that we could use cfg on *all* TLS code, and then this would not happen on Windows.

Reply:
You're right, the problem was on the `StreamSelector` and not `Incoming`. I am OK with leaving this be.